Integrations/integration-jamf.yml

```diff
@@ -30,213 +30,164 @@ configuration:
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    def jamf_get_computers():
+        endpoint = "/JSSResource/computers"
+        try:
+            contents = requests.get(base_url + endpoint,auth=auth,headers=HEADERS).json()
+            ec = {}
+            ec["JAMF"] = [{"ID": item["id"], "Name":item["name"]} for item in contents['computers']]
+
+            demisto.results({ 'Type' : entryTypes['note'], 'ContentsFormat' : formats['json'], 'Contents' : contents, "EntryContext": ec,"HumanReadable": tableToMarkdown("Computers", ec["JAMF"])})
+        except Exception,e:
+            demisto.results({ 'Type' : entryTypes['error'], 'ContentsFormat' : formats['text'], 'Contents' : "Unable to complete command.  Please check credentials and/or proxy settings" })
+            sys.exit(0)
+
+    def jamf_get_computers_match(match):
+        endpoint = "/JSSResource/computers/match/{match}"
+        try:
+            contents = requests.get(base_url + endpoint.format(match=match),auth=auth,headers=HEADERS).json()
+            ec = {}
+            ec["JAMF"] = [
+                {"ID": item["id"],
+                "Name": item["name"],
+                "AltMAC": item["alt_mac_address"],
+                "MAC": item["mac_address"],
+                "Serial": item["serial_number"],
+                "UDID": item["udid"],
+                "AssetTag": item["asset_tag"],
+                "BarCode-1": item["bar_code_1"],
+                "BarCode-2": item["bar_code_2"],
+                "Building": item["building"],
+                "BuildingName": item["building_name"],
+                "Department": item["department"],
+                "DepartmentName": item["department_name"],
+                "Email": item["email"],
+                "EmailAddress": item["email_address"],
+                "Position": item["position"],
+                "Username": item["username"],
+                "Realname": item["realname"],
+                "Room": item["room"]}
+                for item in contents["computers"]
+            ]
+
+            demisto.results({ 'Type' : entryTypes['note'], 'ContentsFormat' : formats['json'], 'Contents' : contents, "EntryContext": ec,"HumanReadable": tableToMarkdown("Computers", ec["JAMF"])})
+        except Exception,e:
+            demisto.results({ 'Type' : entryTypes['error'], 'ContentsFormat' : formats['text'], 'Contents' : "Unable to complete command.  Please check credentials and/or proxy settings" })
+            sys.exit(0)
+
+    def jamf_computers_hardware(computer_id):
+        endpoint = "/JSSResource/computers/id/{computer_id}/subset/hardware"
+
+        try:
+            contents = requests.get(base_url + endpoint.format(computer_id=computer_id),auth=auth,headers=HEADERS).json()
+            ec = {}
+            ec["JAMF"] = {
+                "Make": contents["computer"]["hardware"]["make"],
```

Comment:
your code should be more protective code. don't assume that `hardware` or `make` field will be there. Because if not, the code will throw exception. Instead you can use `dt`, like `"Make": dt(contents, "computer.hardware.make")`